homeassistant/components/systemmonitor/binary_sensor.py

```diff
@@ -140,20 +140,16 @@ def __init__(
 
     async def async_added_to_hass(self) -> None:
         """When added to hass."""
-        if TYPE_CHECKING:
-            assert self.unique_id
         self.coordinator.update_subscribers[
             self.entity_description.add_to_update(self)
-        ].add(self.unique_id)
+        ].add(self.entity_id)
         return await super().async_added_to_hass()
 
     async def async_will_remove_from_hass(self) -> None:
         """When removed from hass."""
-        if TYPE_CHECKING:
-            assert self.unique_id
         self.coordinator.update_subscribers[
             self.entity_description.add_to_update(self)
-        ].discard(self.unique_id)
+        ].discard(self.entity_id)
```

Comment:
Maybe use `.remove` so it blows up if there is a mistake and it somehow is already removed

Reply:
I was thinking the opposite that we wouldn't want it to blow up.
Blow up so it gets noticed so we can fix it you mean?

Reply:
> Blow up so it gets noticed so we can fix it you mean?

Yes exactly. Generally prefer bugs to be obvious and explode as loudly as possible so they get fixed, except when the cost of doing so is not acceptable.